packages/rollup-full-node/test/app/web-rpc-handler.spec.ts

```diff
@@ -457,32 +459,105 @@ describe('Web3Handler', () => {
+          )
+          parsedLogs.length.should.eq(1)
+          parsedLogs[0].signature.should.eq(DUMMY_EVENT_NAME)
+        }
+        it('should return correct logs with #nofilter', async () => {
+          const logs = await httpProvider.getLogs({
+            fromBlock: 'latest',
+            toBlock: 'latest',
+          })
+          verifyEventEmitterLogs(logs)
+        })
+        it('should return correct logs with address filter', async () => {
+          const logs = await httpProvider.getLogs({
+            address: eventEmitter.address,
+          })
+          verifyEventEmitterLogs(logs)
+        })
+        it('should return correct logs with a topics filter', async () => {
+          const dummyTopic =
+            eventEmitterFactory.interface.events[DUMMY_EVENT_NAME].topic
+          const logs = await httpProvider.getLogs({
+            topics: [dummyTopic],
+          })
+          verifyEventEmitterLogs(logs)
+        })
+        it('Should throw throw with proper error for unsupported multi-topic filtering', async () => {
+          const dummyTopic =
+            eventEmitterFactory.interface.events[DUMMY_EVENT_NAME].topic
+          assertAsyncThrowsWithMessage(async () => {
+            await httpProvider.getLogs({
+              topics: [dummyTopic, dummyTopic],
+            })
+          }, 'Unsupported filter parameters')
+        })
+      })
+      describe('Nested contract call events', async () => {
+        let sub
+        let master
+        beforeEach(async () => {
+          const subFactory = new ContractFactory(
+            SubEventEmitter.abi,
+            SubEventEmitter.bytecode,
+            wallet
+          )
+          sub = await subFactory.deploy()
+          const masterFactory = new ContractFactory(
+            MasterEventEmitter.abi,
+            MasterEventEmitter.bytecode,
+            wallet
+          )
+          master = await masterFactory.deploy(sub.address)
+        })
+        it('should return nested contract call events with the correct addresses for each log', async () => {
+          await master.callSubEmitter()
+          const logs = await httpProvider.send(Web3RpcMethods.getLogs, [
+            {
+              fromBlock: 'latest',
+              toBlock: 'latest',
+            },
+          ])
+          logs[0].address.should.eq(master.address)
+          logs[1].address.should.eq(sub.address)
+        })
+        it("should return logs which are the same as a transaction receipt's logs", async () => {
+          const tx = await master.callSubEmitter()
+          const gotLogs = await httpProvider.send(Web3RpcMethods.getLogs, [
+            {
+              fromBlock: 'latest',
+              toBlock: 'latest',
+            },
+          ])
+          const receipt = await httpProvider.send(
+            Web3RpcMethods.getTransactionReceipt,
+            [tx.hash]
+          )
+          gotLogs.should.deep.equal(receipt.logs)
         })
```

Comment:
Could you add one more test here that filters by topic "taco", and makes sure that only 1 log is returned, and it's the right one?